### PR TITLE
sell all the share amount if diff between wanted and share balance is…

### PR DESCRIPTION
**Pull request by mmvsha**
… less than 0.001

### CODE REVIEW

File: src/components/MarketMySharesForm/index.js [hunk 1]
```diff
@@ -87,7 +87,15 @@ class MarketMySharesForm extends Component {
   @autobind
   handleSellShare(shareId, shareAmount) {
     const shareIndex = this.props.marketShares.map(share => share.id).indexOf(shareId)
-    return this.props.sellShares(this.props.market, shareIndex, shareAmount).then(() => this.props.reset())
+    const shareBalance = new Decimal(this.props.marketShares[shareIndex].balance).div(1e18)
+    const selectedSellAmount = new Decimal(shareAmount)
+    const sellAmount = new Decimal(shareBalance)
+      .div(1e18)
+      .sub(selectedSellAmount)
```
Comment (andre-meyer) [inline src/components/MarketMySharesForm/index.js]: I think this value is not in Wei. If it is, can you change the variable name to indicate it's in wei? I think the value passed to handleSellShare is in ETH on Line 306

File: src/components/MarketMySharesForm/index.js [hunk 2]
```diff
@@ -61,7 +56,7 @@ class MarketMySharesForm extends Component {
       const share = marketShares.filter(_share => _share.id === extendedSellId)[0]
       const fullAmount = Decimal(share.balance)
         .div(1e18)
-        .toDP(4, 1)
+        .toDP(2, 1)
```
Comment (andre-meyer) [inline src/components/MarketMySharesForm/index.js]: Why was this changed to 2?

File: src/components/MarketMySharesForm/index.js [hunk 3]
```diff
@@ -87,7 +82,10 @@ class MarketMySharesForm extends Component {
   @autobind
   handleSellShare(shareId, shareAmount) {
     const shareIndex = this.props.marketShares.map(share => share.id).indexOf(shareId)
-    return this.props.sellShares(this.props.market, shareIndex, shareAmount).then(() => this.props.reset())
+    const shareBalance = new Decimal(this.props.marketShares[shareIndex].balance).div(1e18).toDP(2, 1)
+    const selectedSellAmount = new Decimal(shareAmount)
+    const sellAmount = shareBalance.sub(selectedSellAmount).lt(MIN_CONSIDER_VALUE) ? shareBalance : shareAmount
+    return this.props.sellShares(this.props.market, shareIndex, sellAmount).then(() => this.props.reset())
```
Comment (andre-meyer) [inline src/components/MarketMySharesForm/index.js]: can we use `async` + `await` here instead of promises?

File: src/components/MarketMySharesForm/index.js [hunk 4]
```diff
@@ -87,7 +82,10 @@ class MarketMySharesForm extends Component {
   @autobind
   handleSellShare(shareId, shareAmount) {
     const shareIndex = this.props.marketShares.map(share => share.id).indexOf(shareId)
-    return this.props.sellShares(this.props.market, shareIndex, shareAmount).then(() => this.props.reset())
+    const shareBalance = new Decimal(this.props.marketShares[shareIndex].balance).div(1e18).toDP(2, 1)
+    const selectedSellAmount = new Decimal(shareAmount)
```
Comment (andre-meyer) [inline src/components/MarketMySharesForm/index.js]: This in Wei now, yes? ^^

Comment (mmvsha) [reply src/components/MarketMySharesForm/index.js]: @andre-meyer share.balance is in Wei, shareAmount is in ether

File: src/components/MarketMySharesForm/index.js (hunk 2 above)
Comment (mmvsha) [reply src/components/MarketMySharesForm/index.js]: @andre-meyer because we are showing only 2 decimals for `share.balance` in the shares table, but initial value for input is 4 decimals

Comment (mmvsha) [reply src/components/MarketMySharesForm/index.js]: So it was like `5.50` in table, but the initial value for input was, for example, `5.5064`

File: src/components/MarketMySharesForm/index.js (hunk 3 above)
Comment (mmvsha) [reply src/components/MarketMySharesForm/index.js]: @andre-meyer it was using promises, i didn't change anytyhing. I will then :smiley:

Comment (andre-meyer) [reply src/components/MarketMySharesForm/index.js]: Thank you! 👍

File: src/components/MarketMySharesForm/index.js (hunk 4 above)
Comment (andre-meyer) [reply src/components/MarketMySharesForm/index.js]: But then it does `shareBalance (in wei) - shareAmountToSell (in ether)`, right? I think I'm confused

File: src/components/MarketMySharesForm/index.js (hunk 2 above)
Comment (andre-meyer) [reply src/components/MarketMySharesForm/index.js]: Ok, got it

File: src/components/MarketMySharesForm/index.js (hunk 4 above)
Comment (mmvsha) [reply src/components/MarketMySharesForm/index.js]: @andre-meyer `const shareBalance = new Decimal(this.props.marketShares[shareIndex].balance).div(1e18)`
It is divided before subtraction
